examples/static_react_task/webapp/src/app.jsx

```diff
@@ -11,19 +11,76 @@ import ReactDOM from "react-dom";
 import { BaseFrontend, LoadingScreen } from "./components/core_components.jsx";
 import { useMephistoTask } from "mephisto-task";
 
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { error: null, errorInfo: null };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    // Catch errors in any components below and re-render with error message
+      this.setState({
+      error: error,
+      errorInfo: errorInfo
+    })
+    // alert Mephisto worker of a component error
+    alert("error from a component" + error)
+    // pass the error and errorInfo to the backend through /submit_task endpoint
+    this.props.handleError({error:error.message, errorInfo:errorInfo})
+  }
+
+  render() {
+    if (this.state.errorInfo) {
+      // Error path
+      return (
+        <div>
+          <h2>Something went wrong.</h2>
+          <details style={{ whiteSpace: 'pre-wrap' }}>
+            {this.state.error && this.state.error.toString()}
+            <br />
+            {this.state.errorInfo.componentStack}
+          </details>
+        </div>
+      );
+    }
+    // Normally, just render children
+    return this.props.children;
+  }
+}
+
 /* ================= Application Components ================= */
 
 function MainApp() {
+
   const {
     blockedReason,
     blockedExplanation,
     isPreview,
     isLoading,
     initialTaskData,
     handleSubmit,
+    handleFatalError,
     isOnboarding,
   } = useMephistoTask();
 
+
+// Adding event listener instead of using window.onerror prevents the error to be caught twice
+window.addEventListener('error', function (event) {
+  if (event.error.hasBeenCaught !== undefined){
+    return false
+  }
+  event.error.hasBeenCaught = true
+  if (confirm("Do you want to report the error?")) {
+          console.log("You pressed OK!");
+          handleFatalError({errorMsg: event.error.message, error: event.error.stack})
+        } else {
+          console.log("You pressed Cancel!");
+        }
+        return true;
+})
```

Comment:
once this seems to be working correctly, let's move it into `wrap_crowd_source.js` so all tasks automatically inherit this functionality